app/extensions.js

```diff
@@ -222,6 +222,61 @@ let generateTorrentManifest = () => {
   }
 }
 
+// Returns the Chromium extension manifest for the ipfsExtension
+// The ipfsExtension handles ipfs:// and dweb:
+// Analagous to the PDFJS extension, it shows a special UI for that type of resource
+let generateIPFSManifest = () => {
```

Comment:
I don't think you need all this. You can just create a regular manifest file and pass the path to the directory like non-component extensions